src/ucs/memory/rcache.c

```diff
@@ -826,11 +826,14 @@ static void ucs_rcache_before_fork(void)
     pthread_mutex_lock(&ucs_rcache_global_list_lock);
     ucs_list_for_each(rcache, &ucs_rcache_global_list, list) {
         if (rcache->params.flags & UCS_RCACHE_FLAG_PURGE_ON_FORK) {
-            /* Fork will trigger process memory invalidation, thus
-             * - MADV_DONTFORK memory won't be available in child process
-             * - other memory will be detached from physical memory and may
-             *   require reregistration
-             * Invalidate cached regions */
+            /* Fork will trigger process memory invalidation. Cache
+             * invalidation intended to solve following cases:
+             * - MADV_DONTFORK memory use case (e.g IB/MD)
+             *   pages won't be available in child process
+             * - pinned regular memory use case (e.g KNEM/MD)
+             *   mmu invalidation will unpin registered memory regions and it
+             *   should be registered again
+             * - other use cases shouldn't be affected */
```

Comment:
```suggestion
            /* Fork will trigger process memory invalidation. Cache
             * invalidation intended to solve following cases:
             * - Pinned memory with MADV_DONTFORK (e.g IB/MD):
             *   If a registered region shares a page with other allocation in the 
             *   process, that allocation won't be available in child process as expected.
             * - Pinned memory without MADV_DONTFORK (e.g KNEM/MD):
             *   DONTFORK is generally required to avoid registration cache becoming out
             *   of sync with virt-to-phys MMU mapping. We compensate for the absence 
             *   of DONTFORK by removing all registered memory regions, and they could
             *  be registered again on-demand.
             * - Other use cases shouldn't be affected 
             */
```